imports/api/HandScoreCalculator.js

```diff
@@ -0,0 +1,151 @@
+ */
+function jpn_selfdraw_delta(points, fu, bonus, dealerWind, winnerWind, riichiSticks) {
+    let winds = {};
+    Constants.WINDS.forEach(w => {winds[w] = 0});
+    let basicPoints;
+    let nonDealerPays;
+    let dealerPays;
+    let currentBonus = bonus;
+    let individualBonusPayout = Constants.JPN_BONUS_POINTS / 3;
+
+    // Check to see if you have to count basic points
+    if (points < 5) {
+        if ((points === 1 && (fu === 20 || fu === 25)) || (points === 2 && fu === 25)) {
+            throw TypeError("Invalid points/fu combination");
+        } else {
+            // Calculate hand value, if it's above a mangan, cap it there
+            basicPoints = fu * Math.pow(2, 2 + points);
+            basicPoints = basicPoints < Constants.JPN_MANGAN_BASE_POINTS ? basicPoints : Constants.JPN_MANGAN_BASE_POINTS;
+        }
+    } else {
+        basicPoints = manganValue(points);
+    }
+    nonDealerPays = Math.ceil(basicPoints / 100 * (dealerWind == winnerWind ? 2 : 1)) * 100;
+    dealerPays = Math.ceil(basicPoints / 100 * 2) * 100;
+
+    // Everyone loses except the winner
+    Object.keys(winds).forEach(v => winds[v] = -nonDealerPays);
+
+    if (winnerWind == dealerWind) {
+        winds[dealerWind] = (nonDealerPays * 3)
+            + (bonus * Constants.JPN_BONUS_POINTS)
+            + (riichiSticks * Constants.JPN_RIICHI_POINTS);
+    } else {
+        winds[dealerWind] = -(dealerPays + bonus * individualBonusPayout)
+        winds[winnerWind] = dealerPays
+            + (nonDealerPays * 2)
+            + (bonus * Constants.JPN_BONUS_POINTS)
+            + (riichiSticks * Constants.JPN_RIICHI_POINTS);
+    }
+
+     return winds;
+};
+
+/**
+ * Calculates the point difference as the result of a mistaken hand
+ * @param {string} loser - the losing seat true wind
+ * @return {Object} containing the point difference for each seat
+ */
+function jpn_mistake_delta(loser) {
+    let winds = {};
+    Constants.WINDS.forEach(w => {winds[w] = Constants.JPN_MISTAKE_POINTS / 3});
+
+    winds[loser] = -Constants.JPN_MISTAKE_POINTS;
+    return winds;
+};
+
+/**
+ * Calculates the total base points  for high value hands
+ * @param {Number} points - total points in hand
+ * @returns {Number} representing number of base points as the result of a certain point threshold,
+ *                   must be greater than mangan minimum points
+ */
+function manganValue(points) {
+    let multiplier;
+    switch (true) {
+    case (points === 5): multiplier = 1; break;
+    case (points <= 7): multiplier = 1.5; break;
+    case (points <= 10): multiplier = 2; break;
+    case (points <= 12): multiplier = 3; break;
+    // After 13 points is hit, we only see multiples of 13
+    case (points === 13): multiplier = 4; break;
+    case (points === 26): multiplier = 4 * 2; break;
+    case (points === 39): multiplier = 4 * 3; break;
+    case (points === 52): multiplier = 4 * 4; break;
+    case (points === 65): multiplier = 4 * 5; break;
+    }
```

Comment:
Just wondering if we can refactor to:

// After 13 points is hit, we only see multiples of 13
case (points >= 13): multiplier = 4 * (div points 13); break;

I would have put an upper guard on the points (such as (65 => points >= 13)), but I'm assuming these are picked from a drop down menu or guarded elsewhere since there is no guard for syntax errors.

What's originally there is certainly more readable, just wondering if you guys want to take down the line count or something.

Reply:
Line count shouldn't matter. At the end of the day, this is more of a preference thing than anything so I'm fine with either way

Reply:
Same, I'll leave it up to Kevin

Reply:
I honestly think it's clearer this way.

Reply:
Which way lol

Reply:
Yep, what button do I press

Reply:
Yep, what button do I press